### PR TITLE
PHPCS: Bump `testVersion` to PHP 7.0+

### LINKED ISSUE #7595: PHP: Remove 5.6 compatibility code
## Task Description

After increasing our minimum required PHP version to 7.0, we can remove any code that was previously needed to make it work on 5.6.

This includes, but is not limited to, the following areas:

https://github.com/google/web-stories-wp/blob/c4b2f02b98713b17962f283a5f2205f93d56fbce/phpcs.xml.dist#L61-L62

https://github.com/google/web-stories-wp/blob/c4b2f02b98713b17962f283a5f2205f93d56fbce/includes/Infrastructure/ServiceContainer/LazilyInstantiatedService.php#L48-L66

https://github.com/google/web-stories-wp/blob/c4b2f02b98713b17962f283a5f2205f93d56fbce/includes/Infrastructure/Injector/SimpleInjector.php#L369-L409

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

After increasing our minimum required PHP version to 7.0, we can remove any code that was previously needed to make it work on 5.6.

## Relevant Technical Choices

I review these documents and didn't find anything to be an issue. 
https://www.php.net/manual/en/migration70.php
https://www.php.net/manual/en/migration70.new-features.php

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7595

### CODE REVIEW

File: includes/Infrastructure/ServiceContainer/LazilyInstantiatedService.php [hunk 1]
```diff
@@ -55,8 +55,7 @@ public function __construct( callable $instantiation ) {
 	 * @return Service Properly instantiated service.
 	 */
 	public function instantiate() {
-		$instantiation = $this->instantiation; // Because uniform variable syntax not supported in PHP 5.6.
-		$service       = $instantiation();
+		$service       = $this->instantiation();
```
Comment (miina) [inline includes/Infrastructure/ServiceContainer/LazilyInstantiatedService.php]: There's a lint issue here now (alignment).

Additionally, it displays `Call to an undefined method Google\Web_Stories\Infrastructure\ServiceContainer\LazilyInstantiatedService::instantiation()` as a check failure. Should something be changed in the lints, too, for this?

**Comment by miina** [comment]
Note: also found the following comment in the `namespace.php` file:
`Throws an error if the site is running on PHP < 5.6`

Probably this should be changed to 7.0 now? Not sure in which PR.

File: includes/Infrastructure/ServiceContainer/LazilyInstantiatedService.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Infrastructure/ServiceContainer/LazilyInstantiatedService.php]: Not sure I understand this comment. Can you give me some more detail please?

Comment (miina) [reply includes/Infrastructure/ServiceContainer/LazilyInstantiatedService.php]: Sorry for the confusion, see the lint failure: https://github.com/google/web-stories-wp/pull/7969/checks?check_run_id=2847956396#step:10:15

Comment (swissspidy) [reply includes/Infrastructure/ServiceContainer/LazilyInstantiatedService.php]: Maybe this?


```suggestion
		$service = ($this->instantiation)();
```

**Comment by spacedmonkey** [comment]
I have reverted the change. To be honest, I am not sure I see the benefit of changing these changes here. They are not exactly to our codebase, it more the template plugin. This PR is much more simple now.

File: includes/Infrastructure/ServiceContainer/LazilyInstantiatedService.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Infrastructure/ServiceContainer/LazilyInstantiatedService.php]: I have removed the change. It is clear to me this change is adding adding value to us.

**Review by swissspidy — APPROVED** [review]
works for me

**Pull request merged**